### DISCUSSION TITLE
(Possible user incompetence) Gigantic total failed number for filter.

### DISCUSSION THREAD

**Discussion by Pepperspraylaugh** [fail2ban/fail2ban#3081] [Q&A]
My issue is that the "failed filters" list is gigantic and cluttering up logs, and seems to not work alot of the time. 

 <details>
   <summary> OS and fail2ban version </summary>
<pre> -Ubuntu 20.04.2 LTS
   -f2b 0.11.1 </pre>
   </details>

Note that I enabled key auth and disabled password logins for my server.

Running "fail2ban status ssd" returns the following:
<details> 

<pre>sudo fail2ban-client status sshd
Status for the jail: sshd
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     4150
|  `- File list:        /var/log/auth.log
`- Actions
   |- Currently banned: 25
   |- Total banned:     1686
   </pre>
  </details>

My .local jail file looks like this:
<details>

<pre> 
  GNU nano 4.8                                                                                 /etc/fail2ban/jail.local                                                                                           
[DEFAULT]

bantime = 30m
bantime.increment = true
bantime.rndtime = 10m
bantime.factor = 1
bantime.formula = ban.Time * math.exp(float(ban.Count+1)*banFactor)/math.exp(1*banFactor)
maxretry = 2
banaction = iptables-multiport
banaction_allports = iptables-allports

[sshd]

enabled = true
mode = aggressive
port = ssh
logpath = %(sshd_log)s
backend = %(sshd_backend)s
maxretry = 2 
</pre>

</details>

Running "fail2ban-regex /var/log/auth.log ssh" returns:

<details>
<pre>
Running tests
=============

Use   failregex filter file : sshd, basedir: /etc/fail2ban
Use         maxlines : 1
Use      datepattern : Default Detectors
Use         log file : /var/log/auth.log
Use         encoding : UTF-8

Results
=======

Failregex: 41041 total
|-  #) [# of hits] regular expression
|   4) [8038] ^Failed \b(?!publickey)\S+ for (?P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
|   6) [4095] ^[iI](?:llegal|nvalid) user <F-USER>.*?</F-USER> from <HOST>(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  14) [7093] ^<F-NOFAIL>pam_[a-z]+\(sshd:auth\):\s+authentication failure;</F-NOFAIL>(?:\s+(?:(?:logname|e?uid|tty)=\S*)){0,4}\s+ruser=<F-ALT_USER>\S*</F-ALT_USER>\s+rhost=<HOST>(?:\s+user=<F-USER>\S*</F-USER>)?(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  15) [109] ^(error: )?maximum authentication attempts exceeded for <F-USER>.*</F-USER> from <HOST>(?: (?:port \d+|on \S+)){0,2}(?: ssh\d*)?(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*$
|  17) [1] ^<F-MLFFORGET>Disconnecting</F-MLFFORGET>(?: from)?(?: (?:invalid|authenticating)) user <F-USER>\S+</F-USER> <HOST>(?: (?:port \d+|on \S+)){0,2}:\s*Change of username or service not allowed:\s*.*\[preauth\]\s*$
|  19) [10415] ^<F-NOFAIL>Received <F-MLFFORGET>disconnect</F-MLFFORGET></F-NOFAIL> from <HOST>(?: (?:port \d+|on \S+)){0,2}:\s*11:
|  20) [11269] ^<F-NOFAIL><F-MLFFORGET>(Connection closed|Disconnected)</F-MLFFORGET></F-NOFAIL> (?:by|from)(?: (?:invalid|authenticating) user <F-USER>\S+|.+?</F-USER>)? <HOST>(?:(?: (?:port \d+|on \S+|\[preauth\])){0,3}\s*|\s*)$
|  21) [21] ^<F-MLFFORGET><F-MLFGAINED>Accepted \w+</F-MLFGAINED></F-MLFFORGET> for <F-USER>\S+</F-USER> from <HOST>(?:\s|$)
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [45324] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
`-

Lines: 45324 lines, 28798 ignored, 12243 matched, 4283 missed
[processed in 18.34 sec]

Ignored line(s): too many to print.  Use --print-all-ignored to print all 28798 lines
Missed line(s): too many to print.  Use --print-all-missed to print all 4283 lines
</pre>

</details>

**Comment by sebres** [comment]
> Gigantic total failed number for filter

Do you mean "Total failed: 4150"? It is not gigantic at all. I was seeing jails with over 1M found failures (in few days or weeks).
Note that "total failed" is a count of all attempts the filter found since last (re)start of fail2ban. 
The value 4K is nothing in that sense.

> My issue is that the "failed filters" list is gigantic and cluttering up logs, and seems to not work alot of the time.

What is the "failed filters" list? 
What do you mean "it seems to not work alot of the time"?

Neither I see any problem with sshd jail, nor your excerpts illustrate something wrong.
So what is your issue basically?

**Reply by sebres** [reply]
Oh... if you meant "Total failed" shows the number of failures of fail2ban/filter, the assumption is wrong. This is simply a count of all failed tries found in the log (failed logon attempts etc). This does not mean the filter has some errors or has any issue.
This "failures" mean the fails of foreign users found by fail2ban's filter, nothing else.
